components/raftstore/src/store/peer.rs

```diff
@@ -276,19 +276,31 @@ impl<S: Snapshot> CmdEpochChecker<S> {
             .map(|cmd| cmd.index)
     }
 
-    pub fn advance_apply(&mut self, index: u64, term: u64, region: &metapb::Region) {
+    pub fn advance_apply<T>(
+        &mut self,
+        index: u64,
+        term: u64,
+        region: &metapb::Region,
+        find_current_regions: T,
+    ) where
+        T: Fn(&metapb::Region) -> Vec<metapb::Region>,
+    {
         self.maybe_update_term(term);
+        let mut current_regions = None;
         while !self.proposed_admin_cmd.is_empty() {
             let cmd = self.proposed_admin_cmd.front_mut().unwrap();
             if cmd.index <= index {
                 for cb in cmd.cbs.drain(..) {
+                    if current_regions.is_none() {
+                        current_regions = Some(find_current_regions(region));
+                    }
                     let mut resp = cmd_resp::new_error(Error::EpochNotMatch(
                         format!(
                             "current epoch of region {} is {:?}",
                             region.get_id(),
                             region.get_region_epoch(),
                         ),
-                        vec![region.to_owned()],
+                        current_regions.clone().unwrap(),
```

Comment:
🤔 Although the count of merge cmd is far less than split cmd, do we need to determine whether it is a split cmd by adding a cmd type?

Reply:
I think it could help merge cmd too.